### ISSUE TITLE
lib_game_detector only detects steam library.

### ISSUE THREAD

**Issue by g1i7ch** [Rolv-Apneseth/lib_game_detector#6]
I'm not sure if this is an issue, or just my ignorance. 

I'm not able to get lib_game_detector to detect anything other than the steam library using with `rofi-games`. 

I'm posting the issue here, as `rofi-games` seems to be working fine for steam games, so i'm assuming it's a configuration issue on my part or possibly an issue with my environment?

If I understand correctly, as long as the portrait game art is set in lutris, the games should display? i've tried changing to several different variants found at https://www.steamgriddb.com/grids to no avail.

I'm running Arch Linux with Hyprland, currently using the lbonn fork of `rofi`, though ive tried all different variants with no luck.

Is there something i'm missing? Do i need to add a specific library to my PATH?

Any help would be greatly appreciated.

**Comment by Rolv-Apneseth** [comment]
Hi, thanks for opening this.

> If I understand correctly, as long as the portrait game art is set in lutris, the games should display? i've tried changing to several different variants found at https://www.steamgriddb.com/grids to no avail.

In theory yes, it works for me. It is very possible I've made some assumptions that aren't true for all cases though.

First thing that comes to mind, do you have Lutris installed through `pacman` or something else like `flatpak`?

Also, you're setting the image on the left here right?

![epic_lutris](https://github.com/Rolv-Apneseth/lib_game_detector/assets/69486699/75967de2-1606-431b-ac65-b2e90fb2d806)

Otherwise could you run `rofi-games` with the following command and give me the output?

```bash
RUST_LOG=debug rofi -modi games -show games -show-icons
```

**Comment by g1i7ch** [comment]
Thanks for taking the time, and the quick response!

I currently have it installed with pacman, but I've tried the flatpak as well. 

Yes, i'm setting the image on the left. 

`RUST_LOG=debug rofi -modi games -show games -show-icons
DEBUG lib_game_detector::linux::launchers::steam: 173: Steam dir path exists: true
DEBUG lib_game_detector::linux::launchers::heroic::heroic_gog: 69: Heroic GOG installed games json file exists: false
DEBUG lib_game_detector::linux::launchers::heroic::heroic_epic: 27: Heroic Launcher's legendary_library json file exists: false
DEBUG lib_game_detector::linux::launchers::heroic::heroic_amazon: 27: Heroic Launcher's nile_library json file exists: false
DEBUG lib_game_detector::linux::launchers::lutris: 141: Lutris games directory exists: false
DEBUG lib_game_detector::linux::launchers::lutris: 142: Lutris box art directory exists: false
DEBUG lib_game_detector::linux::launchers::lutris: 146: Lutris `game-paths.json` file exists: true
DEBUG lib_game_detector::linux::launchers::bottles: 157: Bottles data directory exists: false
DEBUG lib_game_detector::linux::launchers::bottles: 161: Bottles directory exists: false
DEBUG lib_game_detector::linux::launchers::bottles: 162: Bottles library yaml file exists: false
DEBUG get_detected_games:get_steam_libraries{self=Steam { path_steam_dir: "/home/g1i7ch/.local/share/Steam" }}: lib_game_detector::linux::launchers::steam: 183: Steam libraryfolders.vdf path: "/home/g1i7ch/.local/share/Steam/steamapps/libraryfolders.vdf"
`

**Comment by Rolv-Apneseth** [comment]
I see yeah, maybe I assumed it would always be in one location but it's not. Do you have anything under `~/.config/lutris`? Or maybe `$XDG_CONFIG_HOME/lutris`? Otherwise we're going to have to find where your Lutris config lives.

**Comment by g1i7ch** [comment]
That's a negative. Is that something that has to be manually setup with lutris? it works otherwise.

**Comment by Rolv-Apneseth** [comment]
No, it should be automatic as far as I know. I think it's storing the config somewhere else. We need to find where that is. Can you give me the output of the following command:

```bash
sudo find $HOME -type d -name "lutris"
```

**Comment by Rolv-Apneseth** [comment]
> created the `~/.config/lutris` directory and populated the files...

Oh. What do you mean populated the files?

**Comment by g1i7ch** [comment]
> > created the `~/.config/lutris` directory and populated the files...
> 
> Oh. What do you mean populated the files?

Apologies, I ran lutris to generate the files. :D

**Comment by g1i7ch** [comment]
`sudo find $HOME -type d -name "lutris"`

`/home/g1i7ch/.local/share/lutris`
`/home/g1i7ch/builds/lib_game_detector/tests/file_system_mocks/linux/.cache/lutris`
`/home/g1i7ch/builds/lib_game_detector/tests/file_system_mocks/linux/.config/lutris`
`/home/g1i7ch/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lib_game_detector-0.0.3/tests/file_system_mocks/linux/.cache/lutris`
`/home/g1i7ch/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lib_game_detector-0.0.3/tests/file_system_mocks/linux/.config/lutris`
`/home/g1i7ch/.var/app/net.lutris.Lutris/data/lutris`
`/home/g1i7ch/.var/app/net.lutris.Lutris/cache/lutris`
`/home/g1i7ch/.config/lutris`
/home/g1i7ch/cache.backup/.cache.backup/lutris
/home/g1i7ch/.cache/lutris

**Comment by Rolv-Apneseth** [comment]
So after you created the folder, it added those files by itself? Weird, wonder why it didn't do that until you created the folder.

The box art directory it's talking about should be at `~/.cache/lutris/coverart` or `$XDG_CACHE_HOME/lutris/coverart`

**Comment by Rolv-Apneseth** [comment]
> `/home/g1i7ch/.var/app/net.lutris.Lutris/data/lutris`
> `/home/g1i7ch/.var/app/net.lutris.Lutris/cache/lutris`

I'm assuming these are from the flatpak install? Maybe that's where it's putting things. Can you give me an `ls` of those?

**Comment by Rolv-Apneseth** [comment]
I could potentially just have it check both locations

**Comment by g1i7ch** [comment]
`ls /home/g1i7ch/.var/app/net.lutris.Lutris/data/lutris`                                                    
 .    banners    games         pga.db    runtime
 ..   coverart   lutris.conf   runners  
`ls /home/g1i7ch/.var/app/net.lutris.Lutris/cache/lutris `
 .     game-paths.json   lutris.log   tmp
 ..    gog               shaders      ubisoft
 egs   installer         steam        versions.json

**Comment by g1i7ch** [comment]
> I could potentially just have it check both locations

Sounds viable. how would i make it query both locations?

**Comment by g1i7ch** [comment]
With your help, I've found the solution to my problem. I'm going to go ahead and close.

I had to uninstall my `lutris` games/remove them from library, and delete the `~/.var/app/net.lutris.Lutris/data/lutris` directory that my previous install of `flatpak` had created. Then create the default `pacman` lutris data directory `~/.config/lutris` and restart `lutris`. (This recreates the config files in `~/.config/lutris`)


For some reason unknown to me, the `pacman` version of `lutris` failed to create the default `~/.config/lutris` directory, due to it defaulting to an old `flatpak` install's default data directory `~/.var/app/net.lutris.Lutris/data/lutris`
